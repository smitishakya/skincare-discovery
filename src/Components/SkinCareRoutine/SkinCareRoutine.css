.btn{ 
  font-weight: bold; 
  font-size: 20px;
  border-radius: 8px;
  padding:  12px 28px;
  margin: 5px;
  border: 2px solid black;
  width: 100%;
  text-decoration: none;
}


.step1box{
  background-color: #F7CAC9;
}
.step2box{
  background-color: #FF6F61;
}
.step3box{
  background-color: #9487b6;
}
.step4box{
  background-color: #7FCDCD;
}
.step5box{
  background-color: #DFCFBE;
}

.box{
  list-style-type: none;
  margin: 10px;
  padding: 5px;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
 }
.item-box{
  border-radius: 4px;
  margin: 10px 0px;
  max-width: 400px;
  background-color: white;
  padding: 10px 15px;;
}
.step1item{
  border: 2px solid #D65076;
}
.step2item{
  border: 2px solid #E15D44;
}
.step3item{
  border: 2px solid #9487b6;
}
.step4item{
  border: 2px solid #45B8AC;
}
.step5item{
  border: 2px solid #9F9C99;
}


.price{

  font-weight: bold;
}
.detailshead{
  text-align: left;
  font-weight: bold;

}
.details{
  text-align: center;
  
}

.item-box-img{
  width: 100%;
  height: auto;
}
.SelectBar select{
  font-size: 18px;
  margin: 10px;
  margin-left: 28px;
  border-radius: 8px;
  padding: 10px 5px;
  background-color: 	#FFF8DC;
  color: black;
  border: 2px solid black;
 }


@media screen and (min-width: 730px){
  .SelectBar select { 
    font-size: 24px;
  }
  .btn{    
    font-size: 28px; 
    padding:  15px 30px;
   
  
  }
  .box{
    font-size: 25px;
    margin: 20px;
    padding: 5px;
   }
  .item-box{
    margin: 20px 5px;
    padding: 10px 20px;
  
  }
  

}

@media screen and (min-width: 1026px){

  .btn{
    padding:12px 28px;
    
  }
  .box{

    margin: 10px;
    padding: 5px;
    
   }
  .item-box{
    font-size: 20px;
    border-radius: 4px;
    margin: 20px;
    background-color: white;
    padding: 20px;
  }
  
}