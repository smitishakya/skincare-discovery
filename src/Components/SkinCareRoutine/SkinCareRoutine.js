import React from "react";
import Rating from "../Rating/Rating";
import "./SkinCareRoutine.css";

export default function SkinCareRoutine(props) {
    let skintype = props.match.params.skintype;
    if (skintype === "combination") {
        skintype = "Combination Skin";
    }
    if (skintype === "oily") {
        skintype = "Oily Skin";
    }
    if (skintype === "normal") {
        skintype = "Normal Skin";
    }
    if (skintype === "dry") {
        skintype = "Dry Skin";
    }
    let step1 = props.products
        .filter(product => product.category === "Cleanse")
        .sort((a, b) => a.originalprice - b.originalprice);
    let step2 = props.products
        .filter(product => product.category === "Treat")
        .sort((a, b) => a.originalprice - b.originalprice);
    let step3 = props.products
        .filter(product => product.category === "Tone")
        .sort((a, b) => a.originalprice - b.originalprice);
    let step4 = props.products
        .filter(product => product.category === "Nourish")
        .sort((a, b) => a.originalprice - b.originalprice);
    let step5 = props.products
        .filter(product => product.category === "Hydrate")
        .sort((a, b) => a.originalprice - b.originalprice);
    return (

        <main className="skinCareRoutine">
            <div className="SelectBar">
                <select aria-label="Select" onChange={e => props.onRating(e.currentTarget.value)}>
                    <option value="0">Rating:</option>
                    <option value="5">5</option>
                    <option value="4">4</option>
                    <option value="3">3</option>
                    <option value="2">2</option>
                    <option value="1">1</option>
                </select>
                <select aria-label="Select" onChange={e => props.onRange(e.currentTarget.value)}>
                    <option value="">Range:</option>
                    <option value="$$$">$$$</option>
                    <option value="$$">$$</option>
                    <option value="$">$</option>
                </select>
            </div>
            <button className="step1box btn" onClick={() => props.display1()}>Step 1: Cleanse</button>
            {props.show1 ?
                <ul className="box">
                    {step1
                        .filter(
                            product =>
                                (product.rating >= props.rating) &
                                (product.skintype === skintype) &
                                (props.range === "" ? true : props.range === product.range)
                        )
                        .map((product, index) => (
                            <li className="item-box step1item" key={index}>
                                <img className="item-box-img"
                                    alt="products"

                                    src={
                                        product.image.includes("http") ||
                                            product.image.includes("https")
                                            ? product.image
                                            : `/images/${product.image}`
                                    }
                                />
                                <h1 className="name">{product.name}</h1>
                                <p className="content">{product.content}</p>
                                <Rating value={product.rating} />
                                <p className="price">
                                    {new Intl.NumberFormat("en-US", {
                                        style: "currency",
                                        currency: "USD"
                                    }).format(product.originalprice)}{" "}
                                </p>
                                <p className="detailshead">Details:</p>
                                <p className="details">{product.details}</p>

                            </li>
                        ))}
                </ul>
                : null}
            <button className="step2box btn" onClick={() => props.display2()}>Step 2: Treat</button>
            {props.show2 ?
                <ul className="box">
                    {step2
                        .filter(
                            product =>
                                (product.rating >= props.rating) &
                                (product.skintype === skintype) &
                                (props.range === "" ? true : props.range === product.range)
                        )
                        .map((product, index) => (
                            <li className="item-box step2item" key={index}>
                                <img className="item-box-img"
                                    alt="products"
                                    src={
                                        product.image.includes("http") ||
                                            product.image.includes("https")
                                            ? product.image
                                            : `/images/${product.image}`
                                    }
                                />
                                <h1 className="name">{product.name}</h1>
                                <p className="content">{product.content}</p>
                                <Rating value={product.rating} />
                                <p className="price">
                                    {new Intl.NumberFormat("en-US", {
                                        style: "currency",
                                        currency: "USD"
                                    }).format(product.originalprice)}{" "}
                                </p>
                                <p className="detailshead">Details:</p>
                                <p className="details">{product.details}</p>
                            </li>
                        ))}
                </ul>
                : null}
            <button className="step3box btn" onClick={() => props.display3()}>Step 3: Tone</button>
            {props.show3 ?
                <ul className="box">
                    {step3
                        .filter(
                            product =>
                                (product.rating >= props.rating) &
                                (product.skintype === skintype) &
                                (props.range === "" ? true : props.range === product.range)
                        )
                        .map((product, index) => (
                            <li className="item-box step3item" key={index}>
                                <img className="item-box-img"
                                    alt="products"
                                    src={
                                        product.image.includes("http") ||
                                            product.image.includes("https")
                                            ? product.image
                                            : `/images/${product.image}`
                                    }
                                />
                                <h1 className="name">{product.name}</h1>
                                <p className="content">{product.content}</p>
                                <Rating value={product.rating} />
                                <p className="price">
                                    {new Intl.NumberFormat("en-US", {
                                        style: "currency",
                                        currency: "USD"
                                    }).format(product.originalprice)}{" "}
                                </p>
                                <p className="detailshead">Details:</p>
                                <p className="details">{product.details}</p>
                            </li>
                        ))}
                </ul>
                : null}
            <button className="step4box btn" onClick={() => props.display4()}>Step 4: Nourish</button>
            {props.show4 ?
                <ul className="box">
                    {step4
                        .filter(
                            product =>
                                (product.rating >= props.rating) &
                                (product.skintype === skintype) &
                                (props.range === "" ? true : props.range === product.range)
                        )
                        .map((product, index) => (
                            <li className="item-box step4item" key={index}>
                                <img
                                    alt="products"
                                    src={
                                        product.image.includes("http") ||
                                            product.image.includes("https")
                                            ? product.image
                                            : `/images/${product.image}`
                                    }
                                />
                                <h1 className="name">{product.name}</h1>
                                <p className="content">{product.content}</p>
                                <Rating value={product.rating} />
                                <p className="price">
                                    {new Intl.NumberFormat("en-US", {
                                        style: "currency",
                                        currency: "USD"
                                    }).format(product.originalprice)}{" "}
                                </p>
                                <p className="detailshead">Details:</p>
                                <p className="details">{product.details}</p>
                            </li>
                        ))}
                </ul>
                : null}
            <button className="step5box btn" onClick={() => props.display5()}>Step 5: Hydrate</button>
            {props.show5 ?
                <ul className="box">
                    {step5
                        .filter(
                            product =>
                                (product.rating >= props.rating) &
                                (product.skintype === skintype) &
                                (props.range === "" ? true : props.range === product.range)
                        )
                        .map((product, index) => (
                            <li className="item-box step5item" key={index}>
                                <img className="item-box-img"
                                    alt="products"
                                    src={
                                        product.image.includes("http") ||
                                            product.image.includes("https")
                                            ? product.image
                                            : `/images/${product.image}`
                                    }
                                />
                                <h1 className="name">{product.name}</h1>
                                <p className="content">{product.content}</p>
                                <Rating value={product.rating} />
                                <p className="price">
                                    {new Intl.NumberFormat("en-US", {
                                        style: "currency",
                                        currency: "USD"
                                    }).format(product.originalprice)}{" "}
                                </p>
                                <p className="detailshead">Details:</p>
                                <p className="details">{product.details}</p>
                            </li>
                        ))}
                </ul>
                : null}
        </main>
    );
}
