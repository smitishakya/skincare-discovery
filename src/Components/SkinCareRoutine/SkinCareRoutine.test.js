import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SkinCareRoutine from './SkinCareRoutine';

it('renders without crashing', () => {
  const main = document.createElement('main');
  const props= {
    match: {
        params: {
            skintype: "combination",

        }
    },
    products: []
  }
  ReactDOM.render(<SkinCareRoutine {...props} />, main);
  ReactDOM.unmountComponentAtNode(main);
});


it('renders the UI as expected', () => {
    const props= {
        match: {
            params: {
                skintype: "combination",
    
            }
        },
        products: []
      }
    const tree = renderer
      .create(<SkinCareRoutine  {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});

    

