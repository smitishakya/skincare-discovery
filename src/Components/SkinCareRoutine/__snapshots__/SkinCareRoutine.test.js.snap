// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="skinCareRoutine"
>
  <div
    className="SelectBar"
  >
    <select
      aria-label="Select"
      onChange={[Function]}
    >
      <option
        value="0"
      >
        Rating:
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="1"
      >
        1
      </option>
    </select>
    <select
      aria-label="Select"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Range:
      </option>
      <option
        value="$$$"
      >
        $$$
      </option>
      <option
        value="$$"
      >
        $$
      </option>
      <option
        value="$"
      >
        $
      </option>
    </select>
  </div>
  <button
    className="step1box btn"
    onClick={[Function]}
  >
    Step 1: Cleanse
  </button>
  <button
    className="step2box btn"
    onClick={[Function]}
  >
    Step 2: Treat
  </button>
  <button
    className="step3box btn"
    onClick={[Function]}
  >
    Step 3: Tone
  </button>
  <button
    className="step4box btn"
    onClick={[Function]}
  >
    Step 4: Nourish
  </button>
  <button
    className="step5box btn"
    onClick={[Function]}
  >
    Step 5: Hydrate
  </button>
</main>
`;
